SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -939,6 +939,25 @@ - (void)sendMessage:(TSOutgoingMessage *)message
         }
     }
 
+    NSString *localNumber = [TSAccountManager localNumber];
+    if ([localNumber isEqualToString:recipient.uniqueId]) {
+        if (deviceMessages.count < 1) {
+            DDLogInfo(@"Ignoring sync message without linked devices: %@", [message class]);
+            OWSAssert([message isKindOfClass:[OWSOutgoingSyncMessage class]]);
+
+            dispatch_async([OWSDispatch sendingQueue], ^{
+                [recipient save];
```

Comment:
nit: Can you leave a comment as to why we need to save the recipient here?

Reply:
Done.  I'm trying to emulate the behavior of successful sends.